File: src/main/java/com/wanted/market_api/repository/OrderRepository.java
package com.wanted.market_api.repository;

import com.wanted.market_api.entity.Order;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface OrderRepository extends JpaRepository<Order, Long> {
    Optional<Order> findByIdAndSellerId(Long id, Long memberId);
}

File: src/main/kotlin/wanted/market/order/repository/OrderRepository.kt
package wanted.market.order.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import wanted.market.order.entity.Order
import wanted.market.order.entity.OrderStatus
import java.util.Optional

@Repository
interface OrderRepository : JpaRepository<Order, Long> {

    //n+1 문제 해결하기
    fun findByBuyerIdAndOrderStatus(buyerId: Long, orderStatus: OrderStatus) : List<Order>

    fun findByBuyerIdAndProductId(buyerId: Long, productId: Long) : Optional<Order>
    fun findBySellerIdAndOrderStatus(sellerId: Long, orderStatus: OrderStatus): List<Order>

}